docs/src/main/asciidoc/neo4j-guide.adoc

```diff
@@ -0,0 +1,545 @@
+At the end of this guide, there will be a reactive version, which needs however a 4.0 database version.
+
+== The domain
+
+As with some of the other guise, the application shall manage fruit entities.
+
+[source,java]
+----
+package org.acme.neo4j;
+
+public class Fruit {
+
+    public Long id;
+
+    public String name;
+
+    public Fruit() {
+        // This is needed for the REST-Easy JSON Binding
+    }
+
+    public Fruit(String name) {
+        this.name = name;
+    }
+
+    public Fruit(Long id, String name) {
+        this.id = id;
+        this.name = name;
+    }
+}
+----
+
+== Prerequisites
+
+To complete this guide, you need:
+
+* JDK 1.8+ installed with `JAVA_HOME` configured appropriately
+* an IDE
+* Apache Maven 3.5.3+
+* Access to a Neo4j Database
+* Optional Docker for your system
+
+=== Setup Neo4j
+
+The easiest way to start a Neo4j instance is a locally installed Docker environment.
+
+[source,shell,subs="verbatim,attributes"]
+----
+docker run --publish=7474:7474 --publish=7687:7687 -e 'NEO4J_AUTH=neo4j/secret' neo4j:{neo4j_version}
+----
+
+This starts a Neo4j instance, that publishes it's Bolt port on `7687` and a webinterface on http://localhost:7474.
+
+Have a look at the https://neo4j.com/download/?ref=product[download page] for other options to get started with the product itself.
+
+== Creating the Maven project
+
+First, we need a new project. Create a new project with the following command:
+
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=neo4j-quickstart\
+    -Dextensions="neo4j"
+----
+
+It generates:
+
+* the Maven structure
+* a landing page accessible on `http://localhost:8080`
+* example `Dockerfile` files for both `native` and `jvm` modes
+* the application configuration file
+* an `org.acme.datasource.GreetingResource` resource
+* an associated test
+
+The Neo4j extension has been added already to your pom.
+If you have an already created project, the `neo4j` extension can be added to an existing Quarkus project with the `add-extension` command:
+
+[source,shell]
+----
+mvn quarkus:add-extension -Dextensions="neo4j"
+----
+
+Otherwise, you can manually add this to the dependencies section of your `pom.xml` file:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-neo4j</artifactId>
+</dependency>
+----
+
+=== JSON Binding
+
+We will expose `Fruit` instances over HTTP in the JSON format.
+Consequently, you also need to add the `quarkus-resteasy-jsonb` extension:
+
+[source,shell]
+----
+mvn quarkus:add-extension -Dextensions="resteasy-jsonb"
+----
+
+== Configuring
+
+The Neo4j driver can be configured with standard Quarkus properties:
+
+[source]
+.src/main/resources/application.properties
+----
+quarkus.neo4j.uri = bolt://localhost:7687
+quarkus.neo4j.authentication.username = neo4j
+quarkus.neo4j.authentication.password = secret
```

Comment:
Now looks like the good opportunity to ask a question I have: do you plan to add more properties? I suppose you have a connection pool for instance, that would probably need some configuration knobs?

Reply:
Yes, indeed. 

This is what we have in our Spring Boot Starter
https://github.com/neo4j/neo4j-java-driver-spring-boot-starter/blob/master/neo4j-java-driver-spring-boot-autoconfigure/src/main/java/org/neo4j/driver/springframework/boot/autoconfigure/Neo4jDriverProperties.java

I would not mind porting it in time.

Reply:
Yes, that would be great. We at least need the pool configuration.